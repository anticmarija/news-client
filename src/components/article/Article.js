import React from "react";
import PropTypes from "prop-types";
import { useSelector } from "react-redux";

//TODO - handle refresh article
const Article = () => {
  const { activeArticle } = useSelector((state) => state.article);
  const { title, urlToImage, description } = activeArticle;

  return (
    <div>
      <h1>{title}</h1>
      <img alt={title} src={urlToImage} />
      <p>{description}</p>
    </div>
  );
};

Article.propTypes = {
  article: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    urlToImage: PropTypes.string,
  }),
};

export default Article;
