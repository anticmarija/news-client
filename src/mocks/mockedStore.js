import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
export const initialState = {
  topNews: {},
  countries: {
    supportedCountries: [{ countryId: "gb", countryName: "Gb" }],
  },
  categories: {},
  article: {
    activeArticle: {
      title: "active",
      urlToImage:
        "https://i.dailymail.co.uk/1s/2020/06/17/16/29719992-0-image-a-23_1592408042536.jpg",
      description: "desc",
    },
  },
};

export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
