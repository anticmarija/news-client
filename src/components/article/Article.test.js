import React from "react";
import Article from "./Article";
import { mockedStore } from "../../mocks/mockedStore";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";

describe("Test Article component", () => {
  it("should render correctly without video", () => {
    const store = mockedStore();
    const { container } = render(
      <Provider store={store}>
        <Article />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
