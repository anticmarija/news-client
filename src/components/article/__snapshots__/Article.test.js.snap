// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Article component should render correctly without video 1`] = `
<div>
  <h1>
    active
  </h1>
  <img
    alt="active"
    src="https://i.dailymail.co.uk/1s/2020/06/17/16/29719992-0-image-a-23_1592408042536.jpg"
  />
  <p>
    desc
  </p>
</div>
`;
